Assets/MRTK/SDK/Features/UX/Scripts/StateVisualizer/AnimationTarget.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// Definition class for an animation target, utilized in the State Visualizer component.
+    /// </summary>
+    [Serializable]
+    public class AnimationTarget
+    {
+        [SerializeField]
+        [Tooltip("The target game object for animations.")]
+        private GameObject target;
+
+        /// <summary>
+        /// The target game object for animations.
+        /// </summary>
+        public GameObject Target
+        {
+            get => target;
+            set
+            {
+                if (IsTargetObjectValid(value))
+                {
+                    target = value;
+                }
+                else
+                {
+                    target = null;
+                    Debug.LogError("The Target property can only be a child of this game object, the target was set back to null");
+                }
+            }
+        }
+
+        [SerializeReference]
+        [Tooltip("List of animatable properties for the target game object.  Scale and material color are examples of animatable properties.")]
+        private List<IStateAnimatableProperty> stateAnimatableProperties = new List<IStateAnimatableProperty>();
+
+        /// <summary>
+        /// List of animatable properties for the target game object.  Scale and material color are examples of animatable properties.
+        /// </summary>
+        public List<IStateAnimatableProperty> StateAnimatableProperties
+        {
+            get => stateAnimatableProperties;
+            internal set => stateAnimatableProperties = value;
+        }
+
+        public void SetKeyFrames(AnimationClip animationClip)
+        {
+            foreach (var animatableProperty in StateAnimatableProperties)
+            {
+                animatableProperty.Target = Target;
+                animatableProperty.SetKeyFrames(animationClip);
+            }
+        }
+
+        public void RemoveKeyFrames(string animatablePropertyName, AnimationClip animationClip)
+        {
+            IStateAnimatableProperty animatableProperty = GetAnimatableProperty(animatablePropertyName);
+
+            if (animatableProperty != null)
+            {
+                animatableProperty.RemoveKeyFrames(animationClip);
+            }
+        }
+
+        private IStateAnimatableProperty GetAnimatableProperty(string animatablePropertyName)
+        {
+            return StateAnimatableProperties.Find((prop) => prop.AnimatablePropertyName == animatablePropertyName);
+        }
+
+        private bool IsTargetObjectValid(GameObject target)
```

Comment:
See my previous comment. I think you can reuse `TransformExtensions.FindAncestorComponent<StateVisualizer>(target, includeSelf=true)` rather than needing to re-implement this method. Also try to factor out IsTargetObjectValid into a common class to avoid code duplication.